lib/features/steps/app_automator_steps.rb

```diff
@@ -32,4 +32,23 @@
 # @step_input element_id [String] The locator id
 When("I send the keys {string} to the element {string}") do |keys, element_id|
   $driver.send_keys_to_element(element_id, keys)
+end
+
+# Tests that the given payload value is correct for the target BrowserStack platform.
+# If the step is invoked when a remote BrowserStack device is not in use this step will fail.
+#
+# The DataTable used for this step should have `ios` and `android` in the same row as their expected value:
+#   | android | Java.lang.RuntimeException |
+#   | ios     | NSException                |
+#
+# @step_input field_path [String] The field to test
+# @step_input platform_values [DataTable] A table of acceptable values for each platform
+Then("the event {string} matches the correct platform value:") do |field_path, platform_values|
+  if !defined?($driver) || $driver.nil?
+    fail("This step should only be used if the AppAutomateDriver is present")
```

Comment:
Would be good to have a conversation about Rubocop warning at some point, which I've been trying to resolve generally.  A conversation for another day, though, as I imagine it would involved aligning dev environments to some extent (and what you've done is consistent with the existing code).